modules/nw-allocate-load-balancers-to-subnets-procedure.adoc

```diff
@@ -0,0 +1,121 @@
+
+* The OpenShift installer binary for your target OpenShift version.
+
+* An `install-config.yaml` file.
+
+.Procedure
+
+. Prepare the `install-config.yaml` file:
++
+If you haven't already, generate the installation configuration file using the OpenShift installer:
++
+[source,terminal]
+----
+$ openshift-install create install-config --dir=<your_installation_directory>
+----
++
+This command creates the `install-config.yaml` file in the specified directory.
+
+. Define Subnets and assign roles:
++
+Open the `install-config.yaml` file located in `<your_installation_directory>` using a text editor. You will define your VPC subnets and their designated `roles under platform.aws.vpc.subnets` field.
++
+For each AWS subnet you intend the cluster to use, you will create an entry specifying its ID and a list of roles. Each role is an object with a type key. To designate a subnet for the default Ingress Controller, assign it a role with type: `IngressControllerLB`.
++
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: example.com <1>
+metadata:
+name: my-cluster # Example cluster name
+platform:
+    aws:
+        region: us-east-1 <2>
+        vpc: <3>
+        subnets: <4>
+        - id: subnet-0fcf8e0392f0910d5 <5>
+            roles:
+            - type: IngressControllerLB <6>
+            - type: Bootstrap
+        - id: subnet-0xxxxxxxxxxxxxxza
+            roles:
+            - type: IngressControllerLB 
+        - id: subnet-0fcf8e0392f0910d4 # Private Subnet in AZ us-east-1a
+            roles:
+            - type: ClusterNode <7>
+        - id: subnet-0yyyyyyyyyyyyyyzb 
+            roles:
+            - type: ClusterNode
+pullSecret: '...' <8>
+sshKey: '...' <9>
+----
+<1> Your base domain.
+<2> Your AWS region.
+<3> The vpc object under platform.aws contains the subnets list.
+<4> List of all subnet objects that OpenShift will utilize. Each object defines a subnet id and its roles.
+<5> Replace with your actual AWS Subnet ID. 
+<6> The type: IngressControllerLB role specifically designates this subnet for the default Ingress Controller's LoadBalancer. This subnet must be public.
+<7> The type: ClusterNode role designates this subnet for worker nodes. These are typically private subnets.
+<8> Your pull secret.
+<9> Your SSH key.
++
+For Control Plane load balancers, the subnet IDs in this yaml file will look similar to the following:
+[source,yaml]
+----
+…
+        - id: subnet-0fcf8e0392f0910d6 # Public Subnet for External API LB
+            roles:
+            - type: ControlPlaneExternalLB
+        - id: subnet-0fcf8e0392f0910d7 # Private Subnet for Internal API LB
+            roles:
+            - type: ControlPlaneInternalLB
+…
+----
++
+Ensure you list all necessary subnets, public and private across the AZs, and assign them appropriate roles (`ClusterNode`, `ControlPlaneInternalLB`, `ControlPlaneExternalLB`, `Bootstrap`) according to your cluster architecture.
```

Comment:
This is great. Though, here are a few more points to include:

-  Subnets defines the subnets in an existing VPC and can optionally specify their intended roles. If no roles are specified on any subnet, then the subnet roles are decided automatically. In this case, the VPC must not contain any other non-cluster subnets without the kubernetes.io/cluster/<cluster-id> tag.

- If roles are specified for subnets, each subnet must have at least one assigned role, and the ClusterNode, BootstrapNode, IngressControllerLB, ControlPlaneExternalLB, and ControlPlaneInternalLB roles must be assigned to at least one subnet. However, if the cluster scope is internal, then ControlPlaneExternalLB is not required.